### ISSUE TITLE
puzzlescriptmis rapidly gobbled up all memory

### ISSUE THREAD

**Issue by increpare** [bvoq/mixedaim#28]
In like 10 seconds it had used up > 10 gb memory

repro steps:

* load up this script into puzzlescriptmis
https://www.puzzlescript.net/editor.html?hack=3e1c12177bdb96a655002d8aefba72c2

* place this code in the transform section

``` [magnet]->[]
[target]->[]
choose 10 [  ] -> [ wall ]
choose 1 down [ no obstacle | no obstacle ] -> [ p | q ]

choose 1 option 0.5 down [ no obstacle | no obstacle ] -> [ 2 | 3 ]
+ option 0.5 down [ no obstacle | no obstacle ] -> [ 1 | 4 ]

choose 2 down [ no obstacle no target | no obstacle no target ] -> [ i | o ]
```

let it generate versions of the first level and look at the memory consumption increase rather rapidly

**Comment by bvoq** [comment]
Tested and couldn't replicate. Maybe this has to do with no_action, which needs to be replaced with ```[ action Player ] -> cancel```.
I added an error rule for no_action.

**Comment by bvoq** [comment]
I think this is now fixed in 1.5.
See this commit: https://github.com/bvoq/puzzlescriptmis/commit/05919a2eae2c3d81d15084683464045d233f2b5c
